Source/WebKit/Shared/Cocoa/CoreIPCData.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2023 Apple Inc. All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions
+ * are met:
+ * 1. Redistributions of source code must retain the above copyright
+ *    notice, this list of conditions and the following disclaimer.
+ * 2. Redistributions in binary form must reproduce the above copyright
+ *    notice, this list of conditions and the following disclaimer in the
+ *    documentation and/or other materials provided with the distribution.
+ *
+ * THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS ``AS IS''
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ * THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR
+ * PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS
+ * BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF
+ * THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
+
+#if PLATFORM(COCOA)
+
+#import "DataReference.h"
+
+#import <Foundation/Foundation.h>
+#import <cstdint>
+#import <span>
+#import <wtf/RetainPtr.h>
+
+namespace WebKit {
+
+class CoreIPCData {
+public:
+    CoreIPCData();
+
+    CoreIPCData(CFDataRef d)
+        : m_data()
+        , m_ref(CFDataGetBytePtr(d), CFDataGetLength(d))
+    {
+    }
+
+    CoreIPCData(const IPC::DataReference& d)
+        : m_data(adoptCF(CFDataCreate(0, d.data(), d.size())))
+        , m_ref(CFDataGetBytePtr(m_data.get()), CFDataGetLength(m_data.get()))
+    {
+    }
+
+    RetainPtr<CFDataRef> takeData()
+    {
+        return WTFMove(m_data);
+    };
+
+private:
+    friend struct IPC::ArgumentCoder<CoreIPCData, void>;
+    RetainPtr<CFDataRef> m_data;
+    IPC::DataReference m_ref;
```

Comment:
This actually doesn't need to be stored.  Just make a function like this:
IPC::DataReference asDataReference() { ... }

Then make a constructor that takes a span/DataReference.  Then you only serialize the result of calling this function, and then you don't need to use NotSerialized, which is usually an indication that something needs to be refactored or fixed.

Reply:
A single member now which is an IPC::DataReference